rust/crd/src/lib.rs

```diff
@@ -98,54 +104,97 @@ pub struct SparkApplicationStatus {
 )]
 #[serde(rename_all = "camelCase")]
 pub struct SparkApplicationSpec {
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub version: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub mode: Option<String>,
+    /// Mode: cluster or client. Currently only cluster is supported.
+    pub mode: String,
+
+    /// The main class - i.e. entry point - for JVM artifacts.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub main_class: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub main_application_file: Option<String>,
+
+    /// The actual application file that will be called by `spark-submit`.
+    pub main_application_file: String,
+
+    /// User-supplied image containing spark-job dependencies that will be copied to the specified volume mount.
+    /// See the [examples](DOCS_BASE_URL_PLACEHOLDER/spark-k8s/usage-guide/examples).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub image: Option<String>,
+
+    // no doc - docs in ProductImage struct.
     pub spark_image: ProductImage,
-    /// Name of the Vector aggregator discovery ConfigMap.
+
+    /// Name of the Vector aggregator [discovery ConfigMap](DOCS_BASE_URL_PLACEHOLDER/concepts/service_discovery).
     /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
+    /// Follow the [logging tutorial](DOCS_BASE_URL_PLACEHOLDER/tutorials/logging-vector-aggregator)
+    /// to learn how to configure log aggregation with Vector.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub vector_aggregator_config_map_name: Option<String>,
+
+    /// The job builds a spark-submit command, complete with arguments and referenced dependencies
+    /// such as templates, and passes it on to Spark.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub job: Option<CommonConfiguration<SubmitConfigFragment>>,
+
+    /// The driver role specifies the configuration that, together with the driver pod template, is used by
+    /// Spark to create driver pods.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub driver: Option<CommonConfiguration<RoleConfigFragment>>,
```

Comment:
IMHO driver and executor roles should be mandatory, as you can not have a Spark job without those ;)

Reply:
If the job/driver/executor roles are omitted completely, then Spark defaults to one driver and two executors and the job runs.

Reply:
This is about the simplest job that will run:
```
---
apiVersion: spark.stackable.tech/v1alpha1
kind: SparkApplication
metadata:
  name: spark-examples
spec:
  sparkImage:
    productVersion: "3.5.0"
    pullPolicy: IfNotPresent
  mode: cluster
  mainClass: org.apache.spark.examples.SparkALS
  mainApplicationFile: "local:///stackable/spark/examples/jars/spark-examples.jar"
```

Reply:
I'd therefore be happy to leave them as optional - what do you think?

Reply:
Yeah, should be fine (at the very least for now ;) )